### PR TITLE
doc(google,login) : google login doc

### PR DESCRIPTION
**Pull request by arthur740212**
First draft of the google login document

### CODE REVIEW

File: com.playeveryware.eos/Documentation~/google_signin.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+<a href="/com.playeveryware.eos/README.md"><img src="/com.playeveryware.eos/Documentation~/images/PlayEveryWareLogo.gif" alt="README.md" width="5%"/></a>
+
+# <div align="center">Connect Login with GoogleID using Credential Manager</div>
+---
+
+## Overview
+
+This document describes how to enable Google ID authentication in conjunction with Epic Online Services Plugin for Unity.  
+We followed [Authenticate users with Sign in with Google](https://developer.android.com/identity/sign-in/credential-manager-siwg) and implemented a plugin for the users to easily achieve the feature.
```
Comment (WispyMouse) [inline com.playeveryware.eos/Documentation~/google_signin.md]: Given that we maintain *this plugin*, could we say the name of the plugin that was implemented for this problem?

File: com.playeveryware.eos/Documentation~/google_signin.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
+<a href="/com.playeveryware.eos/README.md"><img src="/com.playeveryware.eos/Documentation~/images/PlayEveryWareLogo.gif" alt="README.md" width="5%"/></a>
+
+# <div align="center">Connect Login with GoogleID using Credential Manager</div>
+---
+
+## Overview
+
+This document describes how to enable Google ID authentication in conjunction with Epic Online Services Plugin for Unity.
```
Comment (WispyMouse) [inline com.playeveryware.eos/Documentation~/google_signin.md]: Given that this article is about "Connect Login", would it be worthwhile to use specific language about the Connect interface rather than "authentication"?

File: com.playeveryware.eos/Documentation~/google_signin.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+## Overview
+
+This document describes how to enable Google ID authentication in conjunction with Epic Online Services Plugin for Unity.  
+We followed [Authenticate users with Sign in with Google](https://developer.android.com/identity/sign-in/credential-manager-siwg) and implemented a plugin for the users to easily achieve the feature.
+
+## Setup
+
+### Plugins
+The feature is included in our samples, so no external plugin is needed.
```
Comment (WispyMouse) [inline com.playeveryware.eos/Documentation~/google_signin.md:15]: nit: Can there be a consistent pattern of blank line after header or no blank line after header?

File: com.playeveryware.eos/Documentation~/google_signin.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+### Build Settings
+To allow the feature, follow the build configuration step below.
```
Comment (WispyMouse) [inline com.playeveryware.eos/Documentation~/google_signin.md]: I think I'd use the phrase "To enable the feature," instead of "To allow"

File: com.playeveryware.eos/Documentation~/google_signin.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+To allow the feature, follow the build configuration step below. 
+Add the following in the dependencies of the project's main gradle file. (The default is `mainTemplate.gradle`) 
+```
+        implementation 'androidx.credentials:credentials-play-services-auth:1.3.0'
+        implementation 'androidx.credentials:credentials:1.3.0'
+        implementation 'com.google.android.libraries.identity.googleid:googleid:1.1.1'
+```
+
+### Configurations
+Configurate `GoogleLoginClientID` and optionally`GoogleLoginNonce` in `AndroidConfig`.
```
Comment (WispyMouse) [inline com.playeveryware.eos/Documentation~/google_signin.md]: I do not know if "Configurate" is a word, but it's a cool thing to say. Might need to be "Configure"

Comment (WispyMouse) [inline com.playeveryware.eos/Documentation~/google_signin.md]: nit: There should be a space after "optionally"

File: com.playeveryware.eos/Documentation~/google_signin.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+> Make sure `GoogleLoginClientID` uses the ClientID of the Web Application type.
+
+## Scripts
+
+### Sample Sign In Scripts
+
+Here is a sample script tested in EOS-Unity-Plugin
+
+```cs
+        private void ConnectGoogleId()
```
Comment (WispyMouse) [inline com.playeveryware.eos/Documentation~/google_signin.md:236]: Is this in the codebase anywhere? Could we direct users to that specific file?

File: com.playeveryware.eos/Documentation~/google_signin.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+            SignInWithGoogleManager signInWithGoogleManager = new();
+
+            signInWithGoogleManager.GetGoogleIdToken((string token, string username) => 
+            {
+                if (string.IsNullOrEmpty(token))
+                {
+                    Debug.LogError("Failed to retrieve Google Id Token");
+                    return;
+                }
+                StartConnectLoginWithToken(ExternalCredentialType.GoogleIdToken, token, username);
```
Comment (WispyMouse) [inline com.playeveryware.eos/Documentation~/google_signin.md:247]: I'm not certain it's clear where "StartConnectLoginWithToken" lives for users. "EOSManager", right?

File: com.playeveryware.eos/Documentation~/google_signin.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+The `GoogleLoginClientID` could be found from `https://console.cloud.google.com/apis/`.  
+Follow [Set up your Google APIs console project](https://developer.android.com/identity/sign-in/credential-manager-siwg#set-google) to set up your project.  
+> [!IMPORTANT]
+> Make sure `GoogleLoginClientID` uses the ClientID of the Web Application type.
+
+## Scripts
+
+### Sample Sign In Scripts
+
+Here is a sample script tested in EOS-Unity-Plugin
```
Comment (WispyMouse) [inline com.playeveryware.eos/Documentation~/google_signin.md]: "EOS-Unity-Plugin" sounds like it's referencing this plugin's name. Or is this meant to refer to a spot?

**Review by WispyMouse — CHANGES_REQUESTED** [review]
I have some nit changes, but the most important one to me is the "Configurate" typo and possibly the code sample at the bottom.

File: com.playeveryware.eos/Documentation~/google_signin.md (hunk 1 above)
Comment (arthur740212) [reply com.playeveryware.eos/Documentation~/google_signin.md]: We don't have name though, let me reword it to tool or utility or script?

File: com.playeveryware.eos/Documentation~/google_signin.md (hunk 6 above)
Comment (arthur740212) [reply com.playeveryware.eos/Documentation~/google_signin.md:236]: Yes UILogin, will add.

File: com.playeveryware.eos/Documentation~/google_signin.md (hunk 7 above)
Comment (arthur740212) [reply com.playeveryware.eos/Documentation~/google_signin.md:247]: That is correct

File: com.playeveryware.eos/Documentation~/google_signin.md (hunk 3 above)
Comment (arthur740212) [reply com.playeveryware.eos/Documentation~/google_signin.md:15]: Totally, how did I allow that.

File: com.playeveryware.eos/Documentation~/google_signin.md (hunk 4 above)
Comment (arthur740212) [reply com.playeveryware.eos/Documentation~/google_signin.md]: That is the word I was looking for, thanks.

File: com.playeveryware.eos/Documentation~/google_signin.md (hunk 5 above)
Comment (arthur740212) [reply com.playeveryware.eos/Documentation~/google_signin.md]: Fixed !

File: com.playeveryware.eos/Documentation~/google_signin.md (hunk 8 above)
Comment (arthur740212) [reply com.playeveryware.eos/Documentation~/google_signin.md]: Yes this plugins name.

File: com.playeveryware.eos/Documentation~/google_signin.md (hunk 7 above)
Comment (arthur740212) [reply com.playeveryware.eos/Documentation~/google_signin.md:247]: Actually no, this is also in `UILogin.cs`, and it calls 
`EOSManager.Instance.StartConnectLoginWithOptions(externalType, token, displayName, ConnectLoginTokenCallback);`

**Review by WispyMouse — APPROVED** [review]

**Review by paulhazen — APPROVED** [review]
